server/apps/archive/common.py

```diff
@@ -540,3 +499,40 @@ def convert_task_attributes_to_objectId(doc):
     if ObjectId.is_valid(task.get(LAST_AUTHORING_DESK, None)) and \
             not isinstance(task.get(LAST_AUTHORING_DESK), ObjectId):
         task[LAST_AUTHORING_DESK] = ObjectId(task.get(LAST_AUTHORING_DESK))
+
+
+def copy_metadata_from_user_preferences(doc, repo_type=ARCHIVE):
+    """
+    Copies following properties: byline, signoff, dateline.located, place from user preferences to doc if the repo_type
+    is Archive.
+
+    About Dateline: Dateline has 3 parts: Located, Date (Format: Month Day) and Source. Dateline can either be simple:
+    Sydney, July 30 AAP - or can be complex: Surat,Gujarat,IN, July 30 AAP -. Date in the dateline is timezone
+    sensitive to the Located.  Located is set on the article based on user preferences if available. If located is not
+    available in user preferences then dateline in full will not be set.
+    """
+
+    if repo_type == ARCHIVE:
+        user = get_user()
+
+        if 'dateline' not in doc:
+            current_date_time = dateline_ts = utcnow()
+            doc['dateline'] = {'date': current_date_time, 'source': ORGANIZATION_NAME_ABBREVIATION, 'located': None,
+                               'text': None}
+
+            if user and user.get('user_preferences', {}).get('dateline:located'):
+                located = user.get('user_preferences', {}).get('dateline:located', {}).get('located')
+                if located:
+                    doc['dateline']['located'] = located
+                    doc['dateline']['text'] = format_dateline_to_locmmmddsrc(located, dateline_ts)
+
+        if BYLINE not in doc and user and user.get(BYLINE):
```

Comment:
just thinking - maybe some of those conditions could be simplified using `doc.setdefault` afterwards